### PR TITLE
[4.0] Fix: Saving menu item fails. Call to a member function validate() on bool. Field "show_associations"

### PR DESCRIPTION
**Pull request by wilsonge**
Pull Request for Issue #23696

Related and already merged fix: #23501

I think this is a better alternative fix to #23700 ?

## Testing Instructions
Install a J4 nightly build of today (Monday, 28 January 2019) or update an older J4 with that package

Go to menu items manager in back-end.

Open Home menu item (type "Articles > Featured Articles")

Save it.

Error like described in #23696 (comment)

Apply patch

Try again to save the menu item.

Should be successful.

### CODE REVIEW

**Comment by SharkyKZ** [comment]
Need the same in `filter()`.

File: libraries/src/Form/Form.php [hunk 1]
```diff
@@ -1199,13 +1199,22 @@ public function validate($data, $group = null)
...
-			// Check for an error.
-			if ($valid instanceof \Exception)
+				// Check for an error.
+				if ($valid instanceof \Exception)
+				{
+					$this->errors[] = $valid;
+					$return         = false;
+				}
+			}
+			elseif (!$fieldObj && $input->get($key))
```
Comment (Fedik) [inline libraries/src/Form/Form.php]: maybe need `$input->has($key)`, because value can be 0 or ''

Comment (wilsonge) [reply libraries/src/Form/Form.php]: Fixed!

**Comment by wilsonge** [comment]
@SharkyKZ fixed :)

**Comment by ReLater** [comment]
I have tested this item :white_check_mark: successfully on c1e7e3bed223060dec7063d28b697131da728bde<br /><br />Saving of articles works, too. After reverting pr https://github.com/joomla/joomla-cms/pull/23501<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23716">issues.joomla.org/tracker/joomla-cms/23716</a>.</sub>

**Comment by infograf768** [comment]
> Saving of articles works, too. After reverting pr #23501

How is that? We still miss merging https://github.com/joomla/joomla-cms/pull/23638
here I get
```
Error
Save failed with the following error: Incorrect integer value: '' for column 'hits' at row 1
```
why revert #23501 ?

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 21026973fd7c813862daeb2ecea5ea648aa1b96c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23716">issues.joomla.org/tracker/joomla-cms/23716</a>.</sub>

**Comment by infograf768** [comment]
@ReLater 
Why reverting #23501 ? I tested here after disabling languagefilter and therefore `show_associations` not displaying when editing article Options grace to #23501 and it works fine.

**Comment by ReLater** [comment]
@infograf768 

> why revert #23501 ?

Because I could create/save Articles, too, after reverting ONLY these lines of #23501: 

```		
// Remove show_associations field if associations is not enabled
		if (!$assoc)
		{
			$form->removeField('show_associations', 'attribs');
		}
```

So, I think that we don't need the additional fix in 23501 anymore with this pr here. Wasn't that one of the purposes of the discussion about "hardcoded field names"?

**Comment by ReLater** [comment]
And the show_associations field is not displayed in a mono-lang page in article edit form.

**Comment by infograf768** [comment]
@ReLater 
Yep, you are right. We can now delete the lines you quote above as the AssocField take care of that.
@wilsonge 
Can you add that to your PR?

**Comment by wilsonge** [comment]
PR Amended

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on b2eca8ef9eb54cc6a5dc2292b5e67a1ff41dc773<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23716">issues.joomla.org/tracker/joomla-cms/23716</a>.</sub>

**Comment by alikon** [comment]
I have tested this item :white_check_mark: successfully on 4a69575f5198692d747013dd0129e5925d9dfdc7<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23716">issues.joomla.org/tracker/joomla-cms/23716</a>.</sub>

**Comment by infograf768** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/23716">issues.joomla.org/tracker/joomla-cms/23716</a>.</sub>

**Pull request merged**